kimchi/src/circuits/polynomials/foreign_field_add/gadget.rs

```diff
@@ -62,22 +74,26 @@ impl<F: PrimeField + SquareRootField> CircuitGate<F> {
         (start_row + circuit_gates.len(), circuit_gates)
     }
 
-    /// Create a single foreign field addition gate
+    /// Create a single foreign field addition gate. This is used for example in the final bound check.
     ///     Inputs
     ///         starting row
     ///     Outputs tuple (next_row, circuit_gates) where
     ///       next_row      - next row after this gate
     ///       circuit_gates - vector of circuit gates comprising this gate
     pub fn create_single_ffadd(
         start_row: usize,
+        operation: FFOps,
```

Comment:
Ah, cool... so this is the one that we can use if we need max flexibility, etc.

document new parameter above.